metaci/build/tests/test_models.py

```diff
@@ -146,6 +148,27 @@ def test_set_commit_status(self):
         build_flow.set_commit_status()
         assert build_flow.build.commit_status == "4"
 
+    def test_set_build_with_release(self):
+
+        build_flow = BuildFlowFactory()
+        build_flow.build.plan.role = "release"
+        build_flow.flow_instance = mock.Mock()
+        project_config = MetaCIUniversalConfig()
+        project_config.config_project = {}
+        project_config.config_project["flows"] = {
+            "rida": {
+                "description": "Generates documentation for robot framework libraries",
+                "steps": {1: {"task": "robot_libdoc"}, 2: {"task": "robot_testdoc"}},
+            }
+        }
+        build_flow.run_flow(
+            project_config=project_config,
+            org_config=ScratchOrgInstanceFactory(),
+        )
+        assert build_flow.build.commit_status == "4"
+        build_flow.build.release = Release(repo=RepositoryFactory())
+        breakpoint()
```

Comment:
@jofsky probably the easiest thing to do would be to make a new method of BuildFlow called _prepare_flow_options and move the logic for building the options there. Then all you need to test it is a BuildFlow that's connected to a Release, and there's no need to actually call run_flow.